From b8c40d25c112371c0b78d31d622ca6ba75e1aa1a Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 2 Jun 2020 21:43:15 +0530
Subject: [PATCH 1368/1852] net: xilinx: axiethernet: Fix swiotlb buffer full
 error

commit 380d7956c9af0181345c9e07f673e831b0c765ec from
https://github.com/Xilinx/linux-xlnx.git

In 10G + mcdma(32-bit) zcu102 design we are seeing swiotlb buffer full
error when MTU is set to 9000. The reason is default software IO TLB
64MB size is not sufficient. Hence in order to make all configurations
work with the default kernel parameters, lower rx BD count to 128
(same as 4.19 kernel version). Based on the need the user can always
override the default tx and rx BD count using ethtool utility.

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Reviewed-by: Harini Katakam <harini.katakam@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index e13f5841b938..edaf82091b39 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -54,7 +54,7 @@
 #endif
 /* Descriptors defines for Tx and Rx DMA */
 #define TX_BD_NUM_DEFAULT		64
-#define RX_BD_NUM_DEFAULT		1024
+#define RX_BD_NUM_DEFAULT		128
 #define TX_BD_NUM_MAX			4096
 #define RX_BD_NUM_MAX			4096
 
-- 
2.31.1

